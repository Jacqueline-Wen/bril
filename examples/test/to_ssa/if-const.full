@main {
.b1:
  cond.0: bool = const true;
  br cond.0 .true .false;
.true:
  a.0: int = const 0;
  b.true: int = undef;
  set b.1 b.true;
  set a.1 a.0;
  jmp .zexit;
.false:
  b.0: int = const 1;
  set b.1 b.0;
  a.false: int = undef;
  set a.1 a.false;
  jmp .zexit;
.zexit:
  a.1: int = get;
  b.1: int = get;
  print a.1;
  ret;
}
