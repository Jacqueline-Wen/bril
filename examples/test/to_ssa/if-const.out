@main {
  b: int = undef;
  a: int = undef;
  cond: bool = undef;
  upsilon cond.b1 cond;
  upsilon a.b1 a;
  upsilon b.b1 b;
.b1:
  b.b1: int = phi;
  a.b1: int = phi;
  cond.b1: bool = phi;
  cond.b1.1: bool = const true;
  upsilon cond.true cond.b1.1;
  upsilon a.true a.b1;
  upsilon b.true b.b1;
  upsilon cond.false cond.b1.1;
  upsilon a.false a.b1;
  upsilon b.false b.b1;
  br cond.b1.1 .true .false;
.true:
  b.true: int = phi;
  a.true: int = phi;
  cond.true: bool = phi;
  a.true.1: int = const 0;
  upsilon cond.zexit cond.true;
  upsilon a.zexit a.true.1;
  upsilon b.zexit b.true;
  jmp .zexit;
.false:
  b.false: int = phi;
  a.false: int = phi;
  cond.false: bool = phi;
  b.false.1: int = const 1;
  upsilon cond.zexit cond.false;
  upsilon a.zexit a.false;
  upsilon b.zexit b.false.1;
  jmp .zexit;
.zexit:
  b.zexit: int = phi;
  a.zexit: int = phi;
  cond.zexit: bool = phi;
  print a.zexit;
  ret;
}
